### PR TITLE
Update merkle-patricia-tree to v4, move account trie-related methods to StateManager

### PR DESCRIPTION
**Pull request by ryanio**
This PR:
* Updates `merkle-patricia-tree` to v4
* Moves account trie-related methods to StateManager: `getCode`, `setCode`, `getStorage`, `setStorage`
   * this was a suggestion from @s1na since he believes the vm was the only consumer of these methods
* Removes various unneeded `promisify`s

### CODE REVIEW

**Comment by holgerd77** [comment]
Just had a first look, phew, nice PR! 😄

File: packages/account/package.json [hunk 1]
```diff
@@ -48,7 +48,6 @@
     "@types/bn.js": "^4.11.6",
     "@types/node": "^11.13.4",
     "@types/tape": "^4.13.0",
-    "merkle-patricia-tree": "^3.0.0",
```
Comment (evertonfraga) [inline packages/account/package.json:51]: 👍

File: packages/block/package.json [hunk 2]
```diff
@@ -43,7 +43,7 @@
     "@ethereumjs/tx": "^2.1.2",
     "@types/bn.js": "^4.11.6",
     "ethereumjs-util": "^7.0.2",
-    "merkle-patricia-tree": "^2.3.2"
+    "merkle-patricia-tree": "^4.0.0"
```
Comment (evertonfraga) [inline packages/block/package.json:46]: 👍

File: packages/block/src/block.ts [hunk 3]
```diff
@@ -254,15 +252,7 @@ export class Block {
...
-    await new Promise((resolve, reject) => {
-      this.txTrie.put(rlp.encode(txIndex), tx.serialize(), (err: any) => {
-        if (err) {
-          reject(err)
-        } else {
-          resolve()
-        }
-      })
-    })
+    await this.txTrie.put(rlp.encode(txIndex), tx.serialize())
```
Comment (evertonfraga) [inline packages/block/src/block.ts:255]: yeah, that's much better :)

File: packages/vm/lib/state/stateManager.ts [hunk 4]
```diff
@@ -482,13 +474,7 @@ export default class DefaultStateManager implements StateManager {
...
-    // should be replaced by account.isEmpty() once updated
-    return (
-      account.nonce.toString('hex') === '' &&
-      account.balance.toString('hex') === '' &&
-      account.codeHash.toString('hex') === KECCAK256_NULL_S
-    )
+    return account.isEmpty()
   }
 
   /**
```
Comment (evertonfraga) [inline packages/vm/lib/state/stateManager.ts:480]: 👍

**Pull request merged**